nixos/modules/services/misc/preload.nix

```diff
@@ -0,0 +1,40 @@
+# flatpak service.
+{ self, config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.preload;
+  preload = pkgs.preload;
+in {
+  meta = { maintainers = preload.meta.maintainers; };
+
+  ###### interface
+  options = {
+    services.preload = { enable = mkEnableOption (lib.mdDoc "preload"); };
+  };
+
+  ###### implementation
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ preload ];
```

Comment:
Do not add the package to `environment.systemPackages` if it will not be used directly by users.